### DISCUSSION TITLE
rabbitmq_mqtt multiple node deployment

### DISCUSSION THREAD

**Discussion by janiu-001** [rabbitmq/rabbitmq-server#5688] [Other] [answered]
Hi expert, 

Would you help to confirm if we could deploy the rabbitmq  cluster with rabbitmq_mqtt plugin on multiple node? 

From the doc, it is not supported. 

<img width="1294" alt="image" src="https://user-images.githubusercontent.com/19970581/187624334-9cc6b84f-a60d-4d65-bc27-b259ebbeea9c.png">

**Comment by michaelklishin** [comment, marked as answer]
It specifically does not support clusters of two nodes (it can be formed but there is no such thing as a majority in a two node clusters).

See this [section in the Clustering guide](https://www.rabbitmq.com/clustering.html#node-count).

In other words, use three or five nodes clusters, with MQTT or any other feature that is based on Raft, e.g. quorum queues or streams.

**Reply by janiu-001** [reply]
so you mean it can only support Odd number, but how to control it for auto scale case

And what the mean of node here, it is a virtual host or it is a Rabbitmq service or a k8s node

**Reply by janiu-001** [reply]
another question how could i enable this plugin with helm, i tried the https://github.com/bitnami/charts/tree/master/bitnami/rabbitmq with plugin param, but it could not work

<img width="825" alt="image" src="https://user-images.githubusercontent.com/19970581/187637051-f2732fa0-91a2-474d-bc29-eff3dd26bf7a.png">

If i start several pod, how could i check if the messages are sent to all of the pods?

**Reply by janiu-001** [reply]
now i can enable the rabbit_mqtt, but i had the following errors, would you help to take a look?

<img width="1041" alt="image" src="https://user-images.githubusercontent.com/19970581/187672915-de8f1ba3-56de-47b8-828e-206db975f9a6.png">

**Reply by mkuratczyk** [reply]
Most likely you have a non-MQTT client connecting to the MQTT port.

**Reply by janiu-001** [reply]
i used mqttx, the following is my configuration
<img width="733" alt="image" src="https://user-images.githubusercontent.com/19970581/187685572-aac78d09-9f1a-4eb5-bd4d-e89b376ed54d.png">

**Reply by janiu-001** [reply]
i would like to double confirm my question, it is possible to support deploy rabbitmq cluster with rabbit_mqtt plugin as long as it is odd number.  Is my understanding right?

**Reply by mkuratczyk** [reply]
yes. the problem is with an even number of nodes, not with having multiple nodes. Feel free to PR the docs to make it clear.

**Reply by mkuratczyk** [reply]
oh, and I think mqttx uses MQTT 5 by default, which we don't support currently. You'd need an MQTT 3 client (probably a flag in mqttx)

**Reply by janiu-001** [reply]
ok,  is the "retained messages" supported for mqtt protocol?

**Reply by mkuratczyk** [reply]
https://rabbitmq.com/mqtt.html#features

**Reply by janiu-001** [reply]
if the number of the node must be a odd number, how could we control it if the scale is automatic, such as scale in/out with the k8s HPA which depend on the cpu and memory

**Reply by mkuratczyk** [reply]
This thread is going in too many directions but a short answer is: automatically scaling RabbitMQ is almost certainly a bad idea. And if you really think you need this, you will need to put a lot of effort into this.

**Reply by janiu-001** [reply]
@mkuratczyk Would you like to confirm one question? It is allowed to deploy the Rabbit free version for commercial usage?

**Reply by mkuratczyk** [reply]
https://www.mozilla.org/en-US/MPL/2.0/FAQ/#use
